core/communication_manager.py

```diff
@@ -102,15 +102,16 @@ def _create_session(self, payload):
         callback(payload)
         logging.info("Kickoff complete.")
 
-    def _schedule_job(self, dmljob_obj):
+    def _schedule_jobs(self, dmljob_obj_arr):
         """
         Helper function that schedules a DML Job through the Scheduler.
         """
         if not self.scheduler:
             raise Exception("Communication Manager needs to be configured first!")
-        if not isinstance(dmljob_obj, DMLJob):
-            raise Exception("Job is not valid!")
-        self.scheduler.add_job(dmljob_obj)
+        for dmljob_obj in dmljob_obj_arr:
+            assert isinstance(dmljob_obj, DMLJob), \
+                "Is not DMLJob, is {}".format(type(dmljob_obj))
```

Comment:
Okay sorry I wasn't clear enough. This new information is good, but what I meant was that it's difficult to tell which of the jobs in the array is not the correct type. For example, if there are 100 jobs in the array, but one of them is actually isn't a DMLJob (for whatever reason). How would you figure out which of the jobs was the fake DMLJob? 

I get that this is just defensive programming, and that the probability we even run into such a situation is low, but in case we do, I think this would save a lot of time.

Tl;dr I would add an index or some kind of identifier so that if this assertion fails, it isn't difficult to figure out which job in the array the assertion failed on.

Reply:
Done (see enumerate)